### ISSUE TITLE
Stop writing logs to filesystem

### ISSUE THREAD

**Issue by j4k0bk** [stanfordnlp/dspy#628]
It annoys me that every time i run my application, even without invoking DSPY, i get assertion.log and openai_usage.log writte to filessytem. This is totally unnecessary. Thank you.

**Comment by arnavsinghvi11** [comment]
Hi @j4k0bk , setting dspy.settings(log_openai_usage=False) should avoid this. However, it's probably a good idea to enable an enable/disable flag in settings for all .log files going forward, since these are getting created upon importing dspy.

**Comment by dbarrundia-tiger** [comment]
Also, if you are trying to import `dspy` in a Read-only file system (i.e. AWS Lambda) you wouldn't even be able to import it

```
  File "/function/.deps/dspy_ai-2.4.0-py3-none-any.whl/dspy/__init__.py", line 1, in <module>
    import dsp
  File "/function/.deps/dspy_ai-2.4.0-py3-none-any.whl/dsp/__init__.py", line 1, in <module>
    from .modules import *
  File "/function/.deps/dspy_ai-2.4.0-py3-none-any.whl/dsp/modules/__init__.py", line 1, in <module>
    from .azure_openai import AzureOpenAI
  File "/function/.deps/dspy_ai-2.4.0-py3-none-any.whl/dsp/modules/azure_openai.py", line 7, in <module>
    handlers=[logging.FileHandler("azure_openai_usage.log")],
  File "/usr/local/lib/python3.11/logging/__init__.py", line 1181, in __init__
    StreamHandler.__init__(self, self._open())
  File "/usr/local/lib/python3.11/logging/__init__.py", line 1213, in _open
    return open_func(self.baseFilename, self.mode,
```

**Comment by okhat** [comment]
I agree here, we shouldn't be writing those files unless they're required

**Comment by KCaverly** [comment]
This also should be fixed with #684 
Logging is set up globally, with config for output type, path, and log level.
